xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java

```diff
@@ -242,32 +242,48 @@ public static Type fromString(String ledgerTypeString) {
     /** The user has requested the withdrawal or deposit, or the exchange has detected an initiated deposit,
      * but the exchange still has to fully process the funding.
      * The funds are not available to the user. The funding request may possibly still be cancelled though. */
-    PROCESSING,
+    PROCESSING("WAIT CONFIFMATION","EMAIL CONFIRMATION","VERIFYING","PENDING_APPROVAL","PENDING"),
 
     /** The exchange has processed the transfer fully and successfully.
      * The funding typically cannot be cancelled any more.
      * For withdrawals, the funds are gone from the exchange, though they may have not reached their destination yet.
      * For deposits, the funds are available to the user. */
-    COMPLETE,
+    COMPLETE("COMPLETED"),
 
-    /** The transfer was cancelled by the user. */
-    CANCELLED,
+    /** The transfer was cancelled either by the user or by the exchange. */
+    CANCELLED("REVOKED","CANCEL","REFUND"),
 
     /** The transfer has failed for any reason other than user cancellation after it was initiated and before it was successfully processed.
      * For withdrawals, the funds are available to the user again. */
-    FAILED,
-
+    FAILED("FAILURE"),
     ;
 
-    public static Status parse(String str) {
-      if (str == null) {
-        return null;
+    private String[] statusArray;
+
+    private static final Map<String, Status> fromString = new HashMap<String, Status>();
+
+    static {
+      for (final Status status : values()){
+        fromString.put(status.toString(), status);
```

Comment:
Move this line *after* the loop below to make code more resilient to future errors (the correct enum value string should overwrite any others).